### ISSUE TITLE
Sophia. Segmentation fault. [1.6.7-560-g360a4aa]

### ISSUE THREAD

**Issue by Quvele** [tarantool/tarantool#1280]
* Tarantool 1.6.7-560-g360a4aa
* tarantool-python 0.5.2
* python 3.5
* ubuntu vivid

Space is `{ engine = 'sophia' }` and has three records.
Tarantool crashes on `space.select(tuple())` or `space.select([2])` 
memtx works without problem.

[Here example sources.](https://gist.github.com/anonymous/f3d91126e0167d577b58)

**Comment by rtsisyk** [comment]
Confirmed on 1.6.7-592-g3fb2936 (latest) using Python 2.
```
#0  sr_compare_u64 (b=<error reading variable: Cannot access memory at address 0x0>, a=0x93fb6e "\001") at sophia/runtime/sr_scheme.c:70
#1  sr_cmpu64 (a=0x93fb6e "\001", asz=14, b=<error reading variable: Cannot access memory at address 0x0>, bsz=<optimized out>, arg=0x954990)
    at sophia/runtime/sr_scheme.c:89
#2  0x000000000042700d in sr_schemecompare (a=0x93fb68 "\006", asize=14, b=0x0, bsize=0, arg=<optimized out>) at sophia/runtime/sr_scheme.c:156
#3  0x00000000005c9291 in sr_compare (s=<optimized out>, s=<optimized out>, bsize=0, b=0x0, asize=14, a=<optimized out>) at sophia/runtime/sr_scheme.h:153
#4  si_qrange (q=0x7fffe6ff7d40) at sophia/index/si_query.c:360
#5  0x00000000005e4976 in si_query (q=q@entry=0x7fffe6ff7d40) at sophia/index/si_query.c:389
#6  0x00000000005eba00 in se_reqread (r=0x7ffff7fa5034) at sophia/environment/se_execute.c:59
#7  0x00000000005efd59 in se_dispatch (s=0x9423b8, t=0x7fffe6ff8970, w=0x9579b0) at sophia/environment/se_scheduler.c:765
#8  se_scheduler (s=s@entry=0x9423b8, w=w@entry=0x9579b0) at sophia/environment/se_scheduler.c:817
#9  0x00000000005f0480 in se_worker (arg=0x9579b0) at sophia/environment/se_scheduler.c:370
#10 0x00007ffff7345284 in start_thread (arg=0x7fffe6ffd700) at pthread_create.c:333
#11 0x00007ffff65ca74d in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:109
```

It seems that sophia_index doesn't handle keys with zero parts properly.
1.6.8-364-gc5a9a91 is OK.

**Comment by rtsisyk** [comment]
Fixed in 1.6.7-593-gc17fa86 / 1.6.8-366-g1d0c49e
Please consider to use 1.6.8 to new sophia deployments (see #1222)
